tests/test_modeling_tf_xlnet.py

```diff
@@ -391,6 +392,69 @@ def test_model_from_pretrained(self):
             model = TFXLNetModel.from_pretrained(model_name)
             self.assertIsNotNone(model)
 
+    # overwrite since `TFXLNetLMHeadModel` doesn't cut logits/labels
+    def test_loss_computation(self):
+        config, inputs_dict = self.model_tester.prepare_config_and_inputs_for_common()
+        for model_class in self.all_model_classes:
+            model = model_class(config)
+            if getattr(model, "hf_compute_loss", None):
+                # The number of elements in the loss should be the same as the number of elements in the label
+                prepared_for_class = self._prepare_for_class(inputs_dict.copy(), model_class, return_labels=True)
+                added_label = prepared_for_class[
+                    sorted(list(prepared_for_class.keys() - inputs_dict.keys()), reverse=True)[0]
+                ]
+                loss_size = tf.size(added_label)
+
+                # `TFXLNetLMHeadModel` doesn't cut logits/labels
+                # if model.__class__ in get_values(TF_MODEL_FOR_CAUSAL_LM_MAPPING):
+                #     # if loss is causal lm loss, labels are shift, so that one label per batch
+                #     # is cut
+                #     loss_size = loss_size - self.model_tester.batch_size
```

Comment:
If I understand correctly, this is the difference between the original test and the overwritten one we have here. We can also rewrite this part in the original test into something like the snippet below, removing the need to overwrite the test (which might prevent issues down the road, if we need to touch this test again).

```python
if model.__class__ in get_values(TF_MODEL_FOR_CAUSAL_LM_MAPPING):
    # `TFXLNetLMHeadModel` doesn't cut logits/labels
    if not 'TFXLNetLMHeadModel' in str(model.__class__):
        loss_size = loss_size - self.model_tester.batch_size
```

@Rocketknight1 WDYT?

Reply:
I did it this way simply because from a previous experience (another PR), @sgugger suggested me to move this kind of model-specific logic to its own test. I think both approaches have their own pros/cons, but I don't have strong opinion on it for this case.

cc @sgugger to see if he has opinion on it.

Reply:
Yes, we try to avoid too much custom logic in the common tests, otherwise they become too quickly unreadable with 100 model architectures, each with its own need.

Reply:
Cool, makes sense 👍 @sgugger Do we have some mechanism to ensure we don't forget to propagate changes to custom tests?

Reply:
Sadly not.